pkg/scannerv4/mappers/mappers.go

```diff
@@ -323,21 +334,17 @@ func toProtoV4VulnerabilitiesMap(ctx context.Context, vulns map[string]*claircor
 		}
 		normalizedSeverity := toProtoV4VulnerabilitySeverity(ctx, v.NormalizedSeverity)
 		name := vulnerabilityName(v)
+		// Determine the related CVE for this vulnerability. This is necessary, as NVD is CVE-based.
+		cve, foundCVE := findName(v, cveIDPattern)
 		// Find the related NVD vuln for this vulnerability name, let it be empty if no
 		// NVD vuln for that name was found.
 		var nvdVuln nvdschema.CVEAPIJSON20CVEItem
 		if nvdCVEs, ok := nvdVulns[v.ID]; ok {
-			if v, ok := nvdCVEs[name]; ok {
+			if v, ok := nvdCVEs[cve]; foundCVE && ok {
 				nvdVuln = *v
-			} else {
-				// Pick the first one as a fallback.
-				for _, v := range nvdCVEs {
-					nvdVuln = *v
-					break
-				}
```

Comment:
> @jvdm I deleted this because it seems like we may potentially link this vulnerability to an incorrect CVSS score.

No, I was going to actually suggest this. This was added as a fallback when advisories (mostly RH and OSV) did not match a CVE name against NVD, and we would try our best effort to populate it; But now that we explicitly return NVDs, it might be a good idea to be strict. Description was being filled by NVD will be missing now for some of those cases, though. But I think that's more accurate than wrong -- I never really understood why we fill NVD data in the original vuln as if it was captured from the data source.

Reply:
Looking at the patch, I think this is even better: Some of the NVD enrichments might have been found in the Description, but we only care for the ones found in the Name, if not, Links.